projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,234 @@
+#include "fan_control_solar.h"
+
+#include <stdbool.h>
+
+#include "data_store.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fault_handler.h"
+#include "fault_monitor.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "ms_test_helpers.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Any defines here
+
+// Declare static test variables
+static GpioAddress s_overtemp_addr;
+static GpioAddress s_fan_fail_addr;
+static GpioAddress s_full_speed_addr;
+static uint16_t s_full_speed_temp_threshold;
+static SolarMpptCount s_mppt_count;
+
+static GpioState s_gpio_state;
+static Event s_e;
+
+static GpioAddress s_dummy;
```

Comment:
The usual practice for these kinds of variables is to declare them locally in each function, that way you don't have to worry about state carrying over between functions (or the weird `s_e` variable name). Also, `s_dummy` won't be needed after you make the `gpio_it_trigger_interrupt` change.